x/wasm/keeper/query_plugins_test.go

```diff
@@ -445,6 +453,40 @@ func TestBankQuerierAllMetadata(t *testing.T) {
 	assert.Equal(t, exp, got)
 }
 
+func TestBankQuerierAllMetadataPagination(t *testing.T) {
+	var requestedPagination *query.PageRequest
+	mock := bankKeeperMock{GetDenomsMetadataFn: func(ctx context.Context, req *banktypes.QueryDenomsMetadataRequest) (*banktypes.QueryDenomsMetadataResponse, error) {
+		// t.Log("test")
+		// t.Log(req == nil)
+		requestedPagination = req.Pagination
+		return &banktypes.QueryDenomsMetadataResponse{
+			Metadatas: []banktypes.Metadata{},
+			Pagination: &query.PageResponse{
+				NextKey: nil,
+			},
+		}, nil
+	}}
+
+	ctx := sdk.Context{}
+	q := keeper.BankQuerier(mock)
+	pagination := &wasmvmtypes.PageRequest{
+		Key:   []byte("key"),
+		Limit: 10,
+	}
+	_, gotErr := q(ctx, &wasmvmtypes.BankQuery{
+		AllDenomMetadata: &wasmvmtypes.AllDenomMetadataQuery{
+			Pagination: &wasmvmtypes.PageRequest{
+				Key:   []byte("key"),
+				Limit: 10,
+			},
+		},
+	})
+	require.NoError(t, gotErr)
+	require.Equal(t, requestedPagination.Key, pagination.Key)
```

Comment:
nit: prefer `assert` over `require` when you don't want to abort the test already. 
For example `require.NoError(t, gotErr)` makes sense as the next asserts would be wrong anyway.
While assert(key) and assert (limit) can both add value to the test output. You would avoid going slice by slice.

Sometimes is more efficient to just compare the captured object with an object of same type. This is often more readable and ensures changes to the object type are covered as well
For example:
exp := &query.PageRequest{...}
assert.Equal(t, exp, got)

nit: the pattern with `testify` is usually: Equal(t, expect, got) . While it works the other way around as well, the output on a failing test can be very misleading.